app/assets/src/components/views/amr_heatmap/AMRHeatmapVis.jsx

```diff
@@ -82,30 +83,38 @@ export default class AMRHeatmapVis extends React.Component {
         );
         if (amrCountForRow != undefined) {
           rowValues.push(amrCountForRow[metric]);
+          maxValue = Math.max(maxValue, amrCountForRow[metric]);
         } else {
           rowValues.push(0);
         }
       });
       heatmapValues.push(rowValues);
     });
-    return heatmapValues;
+    return [heatmapValues, maxValue];
   }
 
   updateHeatmap() {
     const rows = this.getHeatmapLabels();
     const columns = this.state.sampleLabels;
-    const values = this.computeHeatmapValues(rows);
+    const [values, maxValue] = this.computeHeatmapValues(rows);
+    this.props.maxValueUpdater(maxValue);
```

Comment:
This feels like the wrong place to calculate the max value. I would calculate it either in `<AMRHeatmapView>` or `<AMRHeatmapControls>`. The reason this feels like the wrong place is that this component `AMRHeatmapVis` doesn't need to know anything about the max value. It doesn't use it, and the data isn't fetched or stored here either. And it's a little awkward calculating it here, then passing it up to the parent via the `maxValueUpdater` function. It's better to put the logic closer to where it ends up getting used.

I realize it might require a little more duplicated logic to put it in AMRHeatmapView or AMRHeatmapControls (and you'll need to make sure you update the max value whenever samplesData or metric changes), but I think it's worth it to put the logic in a more natural place (for the reasons above). When you have many different variables and logic in a large application, following best practices of putting logic in the place you expect makes the code much easier to navigate.